clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -256,11 +256,13 @@ public void poll(long timeout, long now, PollCondition pollCondition) {
      * nor will it execute any delayed tasks.
      */
     public void pollNoWakeup() {
-        disableWakeups();
-        try {
-            poll(0, time.milliseconds(), null);
-        } finally {
-            enableWakeups();
+        synchronized (this) {
```

Comment:
I was trying to remember since I know I did do it that way at one point. It could have been that we thought we needed `disableWakeups()` anyway for `ConsumerCoordinator.close()`, though we could pass the flag around as well. Maybe I just thought it was ugly.